pkgs/development/web/bun/default.nix

```diff
@@ -0,0 +1,61 @@
+{ stdenvNoCC, callPackage, fetchurl, autoPatchelfHook, unzip, openssl, lib }:
+let
+  dists = {
+    aarch64-darwin = {
+      arch = "aarch64";
+      shortName = "darwin";
+      sha256 = "06kf4dyb8y486r2bbpz8znnaxah9y2hwc9km3cigacfchi4dy28j";
+    };
+
+    x86_64-darwin = {
+      arch = "x64";
+      shortName = "darwin";
+      sha256 = "06q8cajxra8lgyd1ap6iqzqdw31gkvlc6l2ma0mm22y63msgdn8b";
+    };
+
+    x86_64-linux = {
+      arch = "x64";
+      shortName = "linux";
+      sha256 = "038w54yyk94042hg27hfm49w12xsg8wh0skza9l9iybvc9qlvh2i";
+    };
+  };
+  dist = dists.${stdenvNoCC.hostPlatform.system} or (throw "Unsupported system: ${stdenvNoCC.hostPlatform.system}");
+in
+stdenvNoCC.mkDerivation rec {
+  version = "0.1.1";
+  pname = "bun";
+
+  src = fetchurl {
+    url = "https://github.com/Jarred-Sumner/bun-releases-for-updater/releases/download/bun-v${version}/bun-${dist.shortName}-${dist.arch}.zip";
+    sha256 = dist.sha256;
+  };
+
+  sourceRoot = ".";
+  unpackCmd = "unzip bun-${dist.shortName}-${dist.arch}.zip";
+  dontConfigure = true;
+  dontBuild = true;
+  nativeBuildInputs = if stdenvNoCC.isLinux then [ autoPatchelfHook ] else [];
+  buildInputs = [ unzip openssl ];
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm 755 ./bun-${dist.shortName}-${dist.arch}/bun $out/bin
```

Comment:
you are copying the binary as $out/bin instead of $out/bin/bun

Reply:
an `mkdir -p $out/bin` before the install command solves that

Reply:
`install` with `-D` makes the dirs

the issue is missing `bun` at the end (my bad)

`install -Dm 755 ./bun-${dist.shortName}-${dist.arch}/bun $out/bin/bun`